### PR TITLE
Feature/886 signal processing cpp

### PR DESCRIPTION
**Pull request by thibaudteil**
* **Tickets addressed:** bsk-886
* **Review:** By commit  
* **Merge strategy:** Merge (no squash)  

## Description
This PR adds a cpp signal processing capability for low pass filtering. It is the same as the c method, except with a unit test

## Verification
google test to check frequency content of the signal, the reduction of noise, and the setters and getters

## Documentation
None

## Future work
None

### CODE REVIEW

File: src/architecture/utilities/signalProcessing.h [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+ */
+
+#ifndef _SIGNAL_PROCESSING_H_
+#define _SIGNAL_PROCESSING_H_
+
+#include <Eigen/Dense>
```
Comment (codyallard28) [inline src/architecture/utilities/signalProcessing.h]: Does this need Eigen/Dense? I am just remember @patkenneally saying something about the fact that core should be enough

File: src/architecture/utilities/tests/test_signalProcessing.cpp [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+#include "opencv2/core.hpp"
+#include "opencv2/imgproc.hpp"
+#include "opencv2/imgcodecs.hpp"
+#include "opencv2/highgui.hpp"
+#include <gtest/gtest.h>
+#include <random>
+
+const double testAccuracy = 1e-8;
+
+double calculateStandardDeviation(const std::vector<double>& input)
```
Comment (codyallard28) [inline src/architecture/utilities/tests/test_signalProcessing.cpp:27]: I think I found where you got this from: https://www.geeksforgeeks.org/cpp-program-to-find-the-standard-deviation-of-array-elements/

I guess there isn't a c++ library that we have that does this?

**Review by codyallard28 — APPROVED** [review]
Just a few comments.

File: src/architecture/utilities/tests/test_signalProcessing.cpp (hunk 2 above)
Comment (thibaudteil) [reply src/architecture/utilities/tests/test_signalProcessing.cpp:27]: there is not!

File: src/architecture/utilities/signalProcessing.h (hunk 1 above)
Comment (patkenneally) [reply src/architecture/utilities/signalProcessing.h]: Eigen/Core should work

File: src/architecture/utilities/tests/test_signalProcessing.cpp [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+{
+    double sum = 0.;
+    double mean = 0.;
+    double standardDeviation = 0.;
+
+    for (int i = 0; i < input.size(); ++i) {
```
Comment (patkenneally) [inline src/architecture/utilities/tests/test_signalProcessing.cpp]: for (double i : input) {

**Review by patkenneally — APPROVED** [review]

File: src/architecture/utilities/signalProcessing.h (hunk 1 above)
Comment (thibaudteil) [reply src/architecture/utilities/signalProcessing.h]: Done

File: src/architecture/utilities/tests/test_signalProcessing.cpp (hunk 3 above)
Comment (thibaudteil) [reply src/architecture/utilities/tests/test_signalProcessing.cpp]: Done

**Pull request merged**